MapboxNavigation/RouteMapViewController.swift

```diff
@@ -546,6 +556,15 @@ class RouteMapViewController: UIViewController {
             return populated(named)
         }
     }
+    
+    fileprivate func leg(with step: RouteStep) -> RouteLeg? {
+        for leg in route.legs {
+            if leg.steps.contains(step) {
+                return leg
```

Comment:
This is more succinctly written as `route.legs.first { $0.steps.contains(step) }`.